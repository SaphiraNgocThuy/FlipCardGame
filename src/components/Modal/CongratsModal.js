import React from "react";
import PropTypes from "prop-types";
import {
  ModalContainer,
  CongratsText,
  StyledModal,
  DetailContainer,
  StyledImage,
  BigText,
} from "./CongratsModalStyles";
import images from "../../assets";
import CurvedButton from "../Button/CurvedButton";

const CongratsModal = ({ visible, onPress, steps }) => {
  return (
    <StyledModal {...{ visible }}>
      <ModalContainer>
        <DetailContainer>
          <StyledImage source={images.congratulation} />
          <BigText>Congratulation!!</BigText>
          <CongratsText>You win this game by {steps} steps</CongratsText>
          <CurvedButton title="One more game?" {...{ onPress }} />
        </DetailContainer>
      </ModalContainer>
    </StyledModal>
  );
};

CongratsModal.propTypes = {
  visible: PropTypes.bool.isRequired,
  onPress: PropTypes.func.isRequired,
  steps: PropTypes.number.isRequired,
};

export default CongratsModal;
