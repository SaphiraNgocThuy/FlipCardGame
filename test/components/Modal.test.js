import React from 'react';
import renderer from 'react-test-renderer';
import CongratsModal from '../../src/components/Modal/CongratsModal';

it('renders correctly', () => {
  const tree = renderer
    .create(<CongratsModal visible={true} onPress={() => {}} steps={4} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
