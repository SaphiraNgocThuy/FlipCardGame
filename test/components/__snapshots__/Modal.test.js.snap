// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  style={
    Array [
      Object {},
    ]
  }
  transparent={true}
  visible={true}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(0, 0, 0, 0.2)",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
          "shadowColor": "rgba(0, 0, 0, 0.5)",
          "shadowOffset": Object {
            "height": 5,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 10,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderRadius": 30,
            "elevation": 10,
            "paddingBottom": 48,
            "paddingLeft": 48,
            "paddingRight": 48,
            "paddingTop": 48,
          },
        ]
      }
    >
      <Image
        resizeMode="cover"
        source={1}
        style={
          Array [
            Object {
              "height": 100,
              "width": 100,
            },
          ]
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#424c58",
              "fontSize": 40,
              "paddingTop": 32,
            },
          ]
        }
      >
        Congratulation!!
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#424c58",
              "fontSize": 30,
              "paddingBottom": 32,
              "paddingTop": 32,
            },
          ]
        }
      >
        You win this game by 
        4
         steps
      </Text>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#2ac5db",
            "borderRadius": 60,
            "justifyContent": "center",
            "opacity": 1,
            "paddingBottom": 32,
            "paddingLeft": 32,
            "paddingRight": 32,
            "paddingTop": 32,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontSize": 40,
                "fontWeight": "bold",
              },
            ]
          }
        >
          One more game?
        </Text>
      </View>
    </View>
  </View>
</Modal>
`;
